store/postgres/src/dynds/private.rs

```diff
@@ -95,26 +96,34 @@ impl DataSourcesTable {
                 &self.manifest_idx,
                 &self.param,
                 &self.context,
+                &self.causality_region,
             ))
             .order_by(&self.vid)
             .load::<Tuple>(conn)?;
 
         let mut dses: Vec<_> = tuples
             .into_iter()
-            .map(|(block_range, manifest_idx, param, context)| {
-                let creation_block = match block_range.0 {
-                    Bound::Included(block) => Some(block),
-
-                    // Should never happen.
-                    Bound::Excluded(_) | Bound::Unbounded => unreachable!("dds with open creation"),
-                };
-                StoredDynamicDataSource {
-                    manifest_idx: manifest_idx as u32,
-                    param: param.map(|p| p.into()),
-                    context,
-                    creation_block,
-                }
-            })
+            .map(
+                |(block_range, manifest_idx, param, context, causality_region)| {
+                    let creation_block = match block_range.0 {
+                        Bound::Included(block) => Some(block),
+
+                        // Should never happen.
+                        Bound::Excluded(_) | Bound::Unbounded => {
```

Comment:
Since we're dealing with integers, wouldn't `Bound::Excluded(block)` be the same as `Bound::Included(block+1)`? It probably won't happen but isn't really fatal.

Reply:
Well we don't use or plan to use excluded initial bounds, so we can't know what they would mean. And our usage of `lower(block_range)` assumes an included bound, so it's best to blow up if an excluded bound is ever found.